File: src/RepositoryModules.CS/SecondaryLabelCriteria.cs
using System;

namespace RepositoryModules
{
    public class SecondaryLabelCriteria
    {
        public static string PropertyKey { get; } = "15db8aaf-3d90-42e0-9d69-dffcf5a6adc0";

        public SecondaryLabelTypes AcceptableTypes { get; set; } = SecondaryLabelTypes.C3Label;
        
        [Obsolete]
        public bool IsRequired { get; set; }

        public SecondaryNoLabelBehavior NoLabelBehavior { get; set; }

        public SecondaryLabelBehavior ItemNumberEqualsToPrimaryOneBehavior { get; set; }
        public SecondaryLabelBehavior SpecifiedByConditionFileBehavior { get; set; }

        [Obsolete]
        public SecondaryLabelBehavior OtherLabelsBehavior { get; set; }

        SecondaryLabelBehavior _otherNotSingleSymbolLabelsBehavior;
        public SecondaryLabelBehavior OtherNotSingleSymbolLabelsBehavior 
        {
            get
            {
#pragma warning disable CS0612 // 型またはメンバーが旧型式です
                if (OtherLabelsBehavior != SecondaryLabelBehavior.Default) 
                {
                    _otherNotSingleSymbolLabelsBehavior = OtherLabelsBehavior;
                    OtherLabelsBehavior = SecondaryLabelBehavior.Default;
                }
#pragma warning restore CS0612 // 型またはメンバーが旧型式です

                return _otherNotSingleSymbolLabelsBehavior;
            }
            set => _otherNotSingleSymbolLabelsBehavior = value;
        }
    }
}

File: src/RepositoryModules.VB/SecondaryLabelCriteria.vb
Public Class SecondaryLabelCriteria
    Public Shared ReadOnly Property PropertyKey As String = "15db8aaf-3d90-42e0-9d69-dffcf5a6adc0"

    Public Property AcceptableTypes As SecondaryLabelTypes = SecondaryLabelTypes.C3Label

    <Obsolete>
    Public Property IsRequired As Boolean

    Public Property NoLabelBehavior As SecondaryNoLabelBehavior

    Public Property ItemNumberEqualsToPrimaryOneBehavior As SecondaryLabelBehavior
    Public Property SpecifiedByConditionFileBehavior As SecondaryLabelBehavior

    <Obsolete>
    Public Property OtherLabelsBehavior As SecondaryLabelBehavior

    Private _otherNotSingleSymbolLabelsBehavior As SecondaryLabelBehavior
    Public Property OtherNotSingleSymbolLabelsBehavior As SecondaryLabelBehavior
        Get
            If OtherLabelsBehavior <> SecondaryLabelBehavior.Default Then
                _otherNotSingleSymbolLabelsBehavior = OtherLabelsBehavior
                OtherLabelsBehavior = SecondaryLabelBehavior.Default
            End If

            Return _otherNotSingleSymbolLabelsBehavior
        End Get
        Set(ByVal value As SecondaryLabelBehavior)
            _otherNotSingleSymbolLabelsBehavior = value
        End Set
    End Property
End Class
